modules/core/src/smithy4s/http/Metadata.scala

```diff
@@ -151,26 +150,13 @@ object Metadata {
     encoder.encode(a)
 
   /**
-    * Decodes all the fields that are bound to http metadata,
-    * shoving values in a map.
+    * If possible, decode the data from fields that are bound to http metadata.
     */
-  def decodePartial[A](metadata: Metadata)(implicit
-      decoder: PartialDecoder[A]
-  ): Either[MetadataError, MetadataPartial[A]] =
+  def decode[A](metadata: Metadata)(implicit
```

Comment:
I'm surprised no calls to `Metadata.decode` happens outside of test files

Or am I missing something